sw/device/lib/base/memory.c

```diff
@@ -7,6 +7,17 @@
 extern uint32_t read_32(const void *);
 extern void write_32(uint32_t, void *);
 
+// Some symbols below are only defined for device builds. For host builds, we
+// their implementations will be provided by the host's libc implementation.
+//
+// If you are getting missing symbol linker errors for these symbols, it's
+// likely because you have specified `-nostdlib` for a host build. Host builds
+// must be linked against the system libc.
+//
+// This approach is used so that DIFs can depend on `memory.h`, but also be
+// built for host-side software.
+
+#if !defined(HOST_BUILD)
```

Comment:
Is there a reason this:
- Isn't `#ifndef`?
- Can't be one big block, instead of per-function directives?

Reply:
To the first: It's easy to miss the `n`

To the second: It's much easier to not miss the `#if` if it's around an individual function - if it's around some parts, but not all, of a file, it's easier to forget or not correctly move the `#endif`.